### ISSUE TITLE
Notebook questions, post aggregation filters and joins not retained, can't make edits

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#11743]
**Describe the bug**
Using the notebook question builder, if you join a table on a key field that is in the output of an aggregation (see screenshots), it will work and all looks good until you revisit the notebook after a page refresh. After that the question still seems to produce the expected output, but in notebook mode many of the "selectable" components are blank and you can't make any changes. The notebook appears broken but the question still returns data.

Note: After further investigation it's simpler than that, it looks like even a simple post aggregation filter will cause this (see 3rd screenshot).

**Logs**
No useful output in either the javascript console or server logs.

**To Reproduce**
1. Ask a Custom Question > Sample Dataset
2. Select Orders, Count by User ID
3. Then click the join, "Previous Results" to "People" where "ID" = "ID"
4. Can preview, visualise, continue to join tables etc.
5. Save the question
6. Without refreshing the page navigate away from the saved notebook then back to it, everything seems fine.
7. Browser refresh and the notebook is now is a broken state, but the question will continue to return data.
8. Most of the buttons on the notebook no longer work as expected and do nothing

**Expected behavior**
I should be able to view and edit the question in notebook mode.

**Screenshots**
Save question before refresh:
![image](https://user-images.githubusercontent.com/4504437/72581042-d6a53f00-3942-11ea-9852-c3f7a91293fb.png)

Saved question after refresh:
![image](https://user-images.githubusercontent.com/4504437/72581136-300d6e00-3943-11ea-84e2-175c13714bae.png)

Simpler question with just a post aggregate filter causing the same issue:
![image](https://user-images.githubusercontent.com/4504437/72581968-13266a00-3946-11ea-8466-fd879783c8b1.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:71.0) Gecko/20100101 Firefox/71.0",
    "vendor": ""
  },
  "system-info": {
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_101-b13",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_101",
    "java.vm.name": "Java HotSpot(TM) Client VM",
    "java.vm.version": "25.101-b13",
    "os.name": "Windows 7",
    "os.version": "6.1",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "sqlserver",
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "tag": "v0.34.1",
      "date": "2020-01-13",
      "branch": "release-0.34.x",
      "hash": "265695c"
    },
    "settings": {
      "report-timezone": "Pacific/Auckland"
    }
  }
}
```

**Severity**
I need to be able to use the notebook to build more complex questions, this sort of issue stops that from working. For me, post aggregation filters are one of the great additions in notebook mode, this renders it pretty useless for me.

**Additional context**
Combine this with #11334 which still seems to be an issue, I've got significant resistance to using the notebook mode which I really want to love.

**Comment by mazameli** [comment]
Repro'd in master and labeling as a P1 as this breaks the notebook in a pretty common use case.

![image](https://user-images.githubusercontent.com/2223916/72651908-d18fc200-3939-11ea-8648-dd4575ae2a92.png)

**Comment by tlrobinson** [comment]
Duplicate of at least one of #11561 #11495 #11235, Resolved by #11751
